File: Src/Icm.Core/Functions/StepwiseValueLine.cs
using Icm.Collections.Generic.StructKeyStructValue;

namespace Icm.Functions
{


	/// <summary>
	/// Stepwise keyed function with Date key and Double value.
	/// </summary>
	/// <remarks></remarks>
	public class StepwiseValueLine : StepwiseKeyedFunction<System.DateTime, double>
	{

		public StepwiseValueLine(double initialValue, ISortedCollection<System.DateTime, double> coll) : base(initialValue, new DateTotalOrder(), new DoubleTotalOrder(), coll)
		{
		}

		public override IMathFunction<System.DateTime, double> EmptyClone()
		{
			return new StepwiseValueLine(V0(), KeyStore);
		}

	}

}

File: Src2/Icm.Core/Functions/StepwiseValueLine.vb
Imports Icm.Collections.Generic.StructKeyStructValue

Namespace Icm.Functions


    ''' <summary>
    ''' Stepwise keyed function with Date key and Double value.
    ''' </summary>
    ''' <remarks></remarks>
    Public Class StepwiseValueLine
        Inherits StepwiseKeyedFunction(Of Date, Double)

        Public Sub New(ByVal initialValue As Double, ByVal coll As ISortedCollection(Of Date, Double))
            MyBase.New(initialValue, New DateTotalOrder, New DoubleTotalOrder, coll)
        End Sub

        Public Overrides Function EmptyClone() As IMathFunction(Of Date, Double)
            Return New StepwiseValueLine(V0, KeyStore)
        End Function

    End Class

End Namespace
